packages/amazonq/src/api.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GenerateAssistantResponseCommandOutput, GenerateAssistantResponseRequest } from '@amzn/codewhisperer-streaming'
+import { AuthUtil } from 'aws-core-vscode/codewhisperer'
+import { ChatSession } from 'aws-core-vscode/codewhispererChat'
+import { api } from 'aws-core-vscode/amazonq'
+
+export default {
+    chatApi: {
+        async chat(request: GenerateAssistantResponseRequest): Promise<GenerateAssistantResponseCommandOutput> {
+            const chatSession = new ChatSession()
+            return chatSession.chat(request)
+        },
+    },
+    authApi: {
+        async reauthIfNeeded() {
+            if (AuthUtil.instance.isConnectionExpired()) {
+                await AuthUtil.instance.showReauthenticatePrompt()
+            }
+        },
+        async getChatAuthState() {
+            return AuthUtil.instance.getChatAuthState()
+        },
+    },
+} as api
```

Comment:
does `as` force (cast) the object to the interface even if it doesn't conform? is there a different syntax we can use which will add stricter type checking?

Reply:
Just tested and it does error if the incorrect properties are given. You would have to do `as unknown as api` to truly override it.

Reply:
Just switched it to satisfies instead of as which is an even stronger check